Examples/ByFramework/xUnit/FSharp.xUnit/XunitFeature.fs

```diff
@@ -1,13 +1,13 @@
 ﻿module ByFramework.XUnit.Feature
 
-open TickSpec
 open Xunit
+open TickSpec.XunitWiring
 
 let source = AssemblyStepDefinitionsSource(System.Reflection.Assembly.GetExecutingAssembly())
 let scenarios resourceName = source.ScenariosFromEmbeddedResource resourceName |> MemberData.ofScenarios
 
 [<Theory; MemberData("scenarios", "Xunit.FSharp.Addition.feature")>]
-let Addition (scenario : Scenario) = scenario.Action.Invoke()
+let Addition (scenario : XunitSerializableScenario) = source.ScenarioAction(scenario).Invoke()
```

Comment:
can the `Invoke` go away ? i.e. could/should this yield `Async<T>`, `Task<T>` or could the body become `source.RunScenario(scenario)` ?

Reply:
No, Action is a delegate with Invoke method ...
Not sure AssemblyStepDefinitionsSource should have the responsibility to Run Scenarios ... but as you prefer.

Reply:
>but as you prefer.

I don't _prefer_ anything - I'm just a person on the internet typing opinions.

My point is that the surface API that TickSpec.Xunit offers (which is an evolution of some code that I spun up ages ago) might benefit from some polishing. In other words: having people ending up typing `.Invoke()` in every Theory impl is a bit baroque. I'm thus suggesting that we find some way that achieves this without using an `Action` - that might be a helper `module` in `TickSpec.Xunit` (XUnitScenarioRunner?).

You're the one working on it so you'll be best placed to make the exact call - I'm absolutely _not_ trying to dictate what you should do - if you think the suggestion doesnt make sense now, don't do it!

Reply:
Yes, got it, but look, it was 2.4.0 before this PR, so I just put it back to that now, should be fine I hope.

Reply:
No hard feelings, I just don't have a strong preference and need others' opinions myself. 

Even before submitting this PR I was tempted to have AssemblyStepDefinitionsSource.Invoke, however I wondered if the Source has the responsibility to invoke scenarios, or the scenarios should invoke themselves (as before the serializable stuff). One thing is for sure - the Source has the dict of cached scenario Actions ... so I thought the Source should only act as source of scenarios + now actions, but should not be responsible for running them.

I would rather go for scenario.Run(), but as this is not possible then either go for source.ScenarioAction(scenario).Invoke (yes, it is long) or source.RunScenario(scenario) ...

Adding a new XunitScenarioRunner is possible, however it is yet another module/class (if we want to be F#-friendly it should rather be a module), which people have to use ... trying to minimize the API surface here and have as few classes/modules as possible. 

P.S.: I do have quite a lot of years in this industry and have even dictated stuff enough times (wondering afterwards if it was really needed, but a purist is a purist till the end) to not getting anymore so emotional about naming (the hardest problem in IT ;).

Reply:
I'd consider this from the perspective of someone that just wants to use Gherkin and is new to F# and/or .NET - the less concepts like Action and `.Invoke` in their face the better I'd say, hence Something like RunScenario makes more sense to me. Also, if there's a single top level `module` with relevant helpers that one would use from TickSpec.Xunit, that would normally be a first angle of attack.

The current code is 'designed' on the basis of trying to balance:
- not having too much cut and pasted boilerplate
- not introducing new concepts
- still allowing people to adjust things if they end up with more complex usage in their scenario and hence want to do some more abstractions in terms of discovery or running.

But the key assumption is that the code will be there in someone's project ready to F12 around.

Now that it's moving into its own assembly, I'd say the constraints move a little.

Ultimately, if you're happy, I'm happy.

Reply:
@bartelink , what do we do with this one? I am undecided between:
1) source.ScenarioAction(scenario).Invoke
2) source.RunScenario(scenario) 
...

Reply:
It's up to you - I guess the main consideration is that now is the time to change it if it's going to change - the syntax you choose will be pinned in the README until at the very least 3.x ;)

I'm happy whichever way you fall on this

Reply:
ok, changed to option 2